### PR TITLE
Fix user warning produced by apex + scheduler combination

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/841

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/distrib_data_parallel.py [hunk 1]
```diff
@@ -373,6 +373,14 @@ def ddp_train(self, process_idx, model):
             model, optimizers = model.configure_apex(amp, model, self.optimizers, self.amp_level)
             self.optimizers = optimizers
 
+            # Reinitialize optimizer.step properties added by schedulers
+            for lr_scheduler in self.lr_schedulers:
```
Comment (Borda) [inline pytorch_lightning/trainer/distrib_data_parallel.py:377]: maybe we can run this only if `trainer.use_amp`

Comment (SkafteNicki) [reply pytorch_lightning/trainer/distrib_data_parallel.py:377]: this is evaluated inside the `if` statement:
`if self.use_amp and not self.use_native_amp:`
so it should already be taken care of

Comment (Borda) [reply pytorch_lightning/trainer/distrib_data_parallel.py:377]: `use_native_amp` does not need apex...

File: pytorch_lightning/trainer/distrib_data_parallel.py [hunk 2]
```diff
@@ -373,6 +373,14 @@ def ddp_train(self, process_idx, model):
...
+                for optimizer in self.optimizers:
+                    scheduler = lr_scheduler['scheduler']
+                    # check that we dont mix users optimizers and schedulers
+                    if scheduler.optimizer == optimizer:
+                        scheduler.__class__.__mro__[-2].__init__(scheduler, optimizer)
```
Comment (justusschock) [inline pytorch_lightning/trainer/distrib_data_parallel.py:382]: Are you sure, this is always -2? If I have my own fancy scheduler, that also derives from some other class, we could maybe have a different mro?

Comment (SkafteNicki) [reply pytorch_lightning/trainer/distrib_data_parallel.py:382]: Did not think about that, this kind of assumes that everything was just derived *only* from `torch.optim.lr_schedulers._LRScheduler`. You are correct, since the mro of this stupid scheduler:
```
class MyScheduler(torch.optim.lr_scheduler.LambdaLR, torch.utils.data.Dataset):
    pass
```
have mro:
```
(__main__.MyScheduler,
 torch.optim.lr_scheduler.LambdaLR,
 torch.optim.lr_scheduler._LRScheduler,
 torch.utils.data.dataset.Dataset,
 object)
```
where the correct index then would be -3 (or 2). Have an idea how to fix this?

**Comment by s-rog** [comment]
```
/opt/conda/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:122: UserWarning: Seems like `optimizer.step()` has been overridden after learning rate scheduler initialization. Please, make sure to call `optimizer.step()` before `lr_scheduler.step()`. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
```

Getting the same warning using native amp on the latest release (1.2.3). Did this PR remove the warning as well, or just make it so that it can be safely ignored?
